pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -906,3 +915,50 @@ type NomadSDConfig struct {
 	// +optional
 	EnableHTTP2 *bool `json:"enableHTTP2,omitempty"`
 }
+type DockerswarmSDConfig struct {
+	// Address of the docker daemon
+	// +kubebuilder:validation:MinLength=1
+	// +required
+	Host string `json:"host"`
+	// ProxyConfig allows customizing the proxy behaviour for this scrape config.
+	// +optional
+	v1.ProxyConfig `json:",inline"`
+	// TLS configuration to use on every scrape request
+	// +optional
+	TLSConfig *v1.SafeTLSConfig `json:"tlsConfig,omitempty"`
+	// Role of the targets to retrieve. Must be `services`, `tasks`, or `nodes`.
+	// +kubebuilder:validation:Enum=services;tasks;nodes
+	// +required
+	Role Role `json:"role"`
+	// The port to scrape metrics from, when `role` is nodes, and for discovered
+	// tasks and services that don't have published ports.
+	// +optional
+	Port *int `json:"port"`
+	// Optional filters to limit the discovery process to a subset of available
+	// resources.
+	// The available filters are listed in the upstream documentation:
+	// Services: https://docs.docker.com/engine/api/v1.40/#operation/ServiceList
+	// Tasks: https://docs.docker.com/engine/api/v1.40/#operation/TaskList
+	// Nodes: https://docs.docker.com/engine/api/v1.40/#operation/NodeList
+	// +optional
+	Filters []*Filter `json:"filters"`
```

Comment:
```suggestion
	Filters []Filter `json:"filters"`
```

Reply:
I noticed that this is the case for Docker and EC2 too, I noticed the similarity with `[]RelabelConfigs`. But decided to wait for a review before taking a decision.